apps/rxjs-catch-error/src/app/app.component.ts

```diff
@@ -1,74 +1,45 @@
-import { Component, ElementRef, ViewChild } from '@angular/core';
-import { HttpClient } from '@angular/common/http';
-import { Observable, Subscription, concatMap, fromEvent, map } from 'rxjs';
+import { Component, DestroyRef, inject } from '@angular/core';
 import { CommonModule } from '@angular/common';
+import { HttpClient } from '@angular/common/http';
+import { FormsModule } from '@angular/forms';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { Subject, concatMap, map } from 'rxjs';
 
 @Component({
   standalone: true,
-  imports: [CommonModule],
+  imports: [CommonModule, FormsModule],
   selector: 'app-root',
   template: `
     <div class="form-container">
       <span
         >possible values: posts, comments, albums, photos, todos, users</span
       >
     </div>
-    <div class="form-container">
-      <input #inputRef type="text" placeholder="Enter text" />
-      <button #buttonRef>Fetch</button>
-    </div>
+    <form class="form-container" (ngSubmit)="submit$$.next()">
+      <input
+        type="text"
+        placeholder="Enter text"
+        [(ngModel)]="input"
+        name="action" />
+      <button>Fetch</button>
+    </form>
     <div class="response">
       {{ response | json }}
     </div>
   `,
-  styles: [
-    `
-      body {
-        font-family: Arial, sans-serif;
-        display: flex;
-        align-items: center;
-        justify-content: center;
-        height: 100vh;
-        margin: 0;
-      }
-      .form-container {
-        text-align: center;
-      }
-      input {
-        padding: 8px;
-        margin-right: 8px;
-        border: 1px solid #ccc;
-        border-radius: 4px;
-      }
-      button {
-        padding: 8px 16px;
-        background-color: #007bff;
-        color: #fff;
-        border: none;
-        border-radius: 4px;
-        cursor: pointer;
-      }
-      .response {
-        margin-left: 25%;
-        margin-top: 2%;
-        width: 50%;
-        text-align: center;
-        border: 1px solid #ccc;
-      }
-    `,
-  ],
+  styleUrls: ['./app.component.css'],
 })
 export class AppComponent {
-  @ViewChild('buttonRef', { static: true }) buttonRef!: ElementRef;
-  @ViewChild('inputRef', { static: true }) inputRef!: ElementRef;
+  submit$$ = new Subject<void>();
+  input = '';
   response: unknown;
+  private destroyRef = inject(DestroyRef);
   constructor(private http: HttpClient) {}
-
   ngOnInit() {
-    const buttonClick$ = fromEvent(this.buttonRef.nativeElement, 'click');
-    buttonClick$
+    this.submit$$
       .pipe(
-        map(() => this.inputRef.nativeElement.value),
+        takeUntilDestroyed(this.destroyRef),
```

Comment:
nit: should be placed as the last operator so all obs are destroyed even the one inside high order operators. In this case, this is not such a big issue since http request completes on their own, but can lead to tricky bug otherwise